### ISSUE TITLE
Godot always import collision shapes as concave no matter if it is concave or convex

### ISSUE THREAD

**Issue by RepairUnit3k6** [godotengine/godot#31760]
**Godot version:**
Godot 3.1.1 Official Stable, Blender 2.79b, 

**OS/device including version:**
LMDE with XFce, Lenovo B50-80 Laptop

**Issue description:**
(3D) Using `-colonly` suffix to import mesh as collision shape, Godot always import it as concave no matter if is shape concave or convex. This problem is even more irritating by fact that collision between two concave shapes is not possible. Some kind of detection is needed to import mesh as convex collision shape if possible. Model was exported into Godot using Blender ESCN export plugin

**Steps to reproduce:**
Extract test project. You can see .blend file and folder with project. In project is test scene, imported .blend file and scene inhered from it. Despite shape is very basic 3D cube made of 12 triangles which is convex shape, Godot imported it as concave thus making collisions with other imported shapes impossible

**Minimal reproduction project:**
[test_project.zip](https://github.com/godotengine/godot/files/3555032/test_project.zip)

**Comment by groud** [comment]
Isn't that a problem with the Blender exporter instead ?

**Comment by Garrom** [comment]
> Isn't that a problem with the Blender exporter instead ?

I thought Godot converts it into nodes... But maybe ? I'll try old better dea exporter

**Comment by Garrom** [comment]
> Isn't that a problem with the Blender exporter instead ?

I tried to use better dae exporter. However, observed same behavior which leads me to believe it indeed is problem with Godot's import feature

**Comment by Cykyrios** [comment]
I ran into this issue recently as well, and tested different options while exporting. My test case was made on master (3.2 alpha) on Linux Mint, where I tried to export a cube with a beveled edge (which is a convex shape). I have the visual mesh named mesh, and the collision mesh named cube. I used the glTF format when exporting.
* Exporting only the visual mesh as mesh-col results in a visual mesh with an exact copy as a ConcavePolygonShape.
* Exporting the visual mesh as is, and the collision mesh as cube-colonly (as a child or sibling of the visual mesh) results in a separate StaticBody with a ConcavePolygonShape which is exactly the same as the visual mesh.
* Same as above, but with cube-convcolonly, results in a ConvexPolygonShape; however it is automatically processed by the VHACD convex decomposition on import, which results in a slightly deformed collision shape (even though it is still only one shape).
* I also ran a different test with a slightly more complex mesh and a few cubes and cylinders as collision meshes, and exporting these with -convcolonly resulted in a rather imprecise (overly conservative) convex decomposition. The same is true when using the convex decomposition option in the editor.

I believe the -convcolonly hint should check for convexity before trying to decompose the mesh, and agree that -colonly should also do that to avoid getting concave shapes everywhere.
I also think the user should have control over the convex decomposition, although that is a separate issue.
